pylint/checkers/refactoring.py

```diff
@@ -261,13 +266,42 @@ def _check_redefined_argument_from_local(self, name_node):
                                  args=(name_node.name, ))
 
     @utils.check_messages('redefined-argument-from-local',
-                          'too-many-nested-blocks')
+                          'too-many-nested-blocks',
+                          'mutable-sequence-modified-in-loop')
     def visit_for(self, node):
         self._check_nested_blocks(node)
 
         for name in node.target.nodes_of_class(astroid.AssignName):
             self._check_redefined_argument_from_local(name)
 
+        if self._check_iter_is_infer_list(node):
+            for part in node.nodes_of_class(astroid.Expr):
+                if self._check_part_is_call_method_elimination(node, part):
+                    self.add_message('mutable-sequence-modified-in-loop',
+                                     node=part,
+                                     args=(node.iter.name))
+
+    @staticmethod
+    def _check_iter_is_infer_list(node):
+        """ Check if node.iter is astroid.util.Uninferable o
+            astroid.node_classes.List instance and if node.body
+            contains records """
+        inferred = utils.safe_infer(node.iter)
+        return (node.body and isinstance(node.iter, astroid.Name) and
+                (inferred is astroid.util.Uninferable or
+                isinstance(inferred, astroid.node_classes.List)))
+
+    @staticmethod
+    def _check_part_is_call_method_elimination(node, part):
+        """ Check if the part is a astroid.Call instance and the method called
+            is in ('remove', 'delete', 'pop') """
+        return (isinstance(part.value, astroid.Call) and
```

Comment:
I think this would be clearer if we separate this check in multiple implementations for each mutable sequence. Also, let's split this long ``if`` in shorter variables, as in:
```
attribute = part.value.func
attrname = attribute.attrname
expr = attribute.expr
``` etc. This will make it more clearer than now.